native/swift/Sources/wordpress-api/WordPressAPI.swift

```diff
@@ -138,6 +143,22 @@ public actor WordPressAPI {
         self.requestExecutor = executor
     }
 
+    public func asSelfHostedService() throws -> WpSelfHostedService {
+        let cache = try WpApiCache(path: nil)
+        _ = try cache.performMigrations()
```

Comment:
Also, based on a hunch (we can probably write some unit tests to verify this), we should not call `performMigrations` on the same database file. I don't think that the migration implementation is supposed to handle multiple `performMigrations` calls at the same time.

I'm aware that this exact code uses an in-memory database, so the above won't be an issue. But we probably should add TODO or FIXME comment to say we are not supposed to have multiple `WpApiCache` instances and call `performMigrations` multiple times.